frontend/src/services/effects.ts

```diff
@@ -180,14 +182,44 @@ export function withEffects(store: Store) {
    * service area too.
    */
   store.on('selectedCounties').subscribe(selectedCounties => {
-    if (selectedCounties === null) {
-      store.set('selectedServiceAreas')(null)
-    } else {
-      let selectedServiceAreas = filter(store.get('serviceAreas'), function (sA) { return selectedCounties.includes(countyFromServiceArea(sA)) })
+    if (selectedCounties !== null) {
+      let selectedServiceAreas = filter(store.get('serviceAreas'), function (sA) {
+        return selectedCounties.includes(parseSerializedServiceArea(sA).county)
+      })
+      store.set('selectedServiceAreas')(selectedServiceAreas)
+    }
+  })
+
+  /**
+   * If the user selects a county, then deselects that service area's
+   * zip code or county in the Service Area drawer, we should de-select the
+   * service area too.
+   */
+  store.on('selectedCountyType').subscribe(selectedCountyType => {
+    if (selectedCountyType !== null) {
+      let selectedServiceAreas = filter(store.get('serviceAreas'), function (sA) {
+        let { state, county } = parseSerializedServiceArea(sA)
+        let nhcs_code = getPropCaseInsensitive(ZIPS_BY_COUNTY_BY_STATE[state], county).nhcs_code
+        let urban = nhcs_code <= 4
+        console.log({ state, county, urban })
+        return (selectedCountyType === 'Urban') ? urban : !urban
+      })
       store.set('selectedServiceAreas')(selectedServiceAreas)
     }
   })
 
+  /**
+   * If the user selects a new selector method, re-select all service areas.
```

Comment:
we should be consistent about what the first step is called. Is it a filterSelector or a selectorMethod?

I think selectorMethod doesn't actually make much sense.

can we call it countySelectorMethod?